src/main/java/com/nike/cerberus/auth/connector/okta/OktaAuthHelper.java

```diff
@@ -0,0 +1,179 @@
+            authResult = this.authClient.authenticateWithFactor(stateToken, factorId, passCode);
+        } catch (IOException ioe) {
+            final String msg = String.format("stateToken: %s failed to verify 2nd factor for reason: %s",
+                    stateToken, ioe.getMessage());
+
+            throw ApiException.newBuilder()
+                    .withApiErrors(DefaultApiError.AUTH_BAD_CREDENTIALS)
+                    .withExceptionMessage(msg)
+                    .build();
+        }
+
+        return authResult;
+    }
+
+    /**
+     * Authenticate a user with Okta
+     *
+     * @param username   Okta username
+     * @param password   Okta password
+     * @param relayState Deep link to redirect user to after authentication
+     * @return Session login token
+     */
+    protected AuthResult authenticateUser(final String username, final String password,
+                                          final String relayState) {
+
+        try {
+            return this.authClient.authenticate(username, password, relayState);
+        } catch (IOException ioe) {
+            final String msg = String.format("failed to get user groups for reason: %s", ioe.getMessage());
+
+            throw ApiException.newBuilder()
+                    .withApiErrors(DefaultApiError.GENERIC_BAD_REQUEST)
+                    .withExceptionMessage(msg)
+                    .build();        }
+    }
+
+    /**
+     * Convenience method for parsing the Okta response and mapping it to a class.
+     *
+     * @param authResult  The Okta authentication result object
+     * @return Deserialized object from the response body
+     */
+    protected EmbeddedAuthResponseDataV1 getEmbeddedAuthData(final AuthResult authResult) {
+
+        Preconditions.checkArgument(authResult != null, "auth result cannot be null.");
+
+        final Map<String, Object> embedded = authResult.getEmbedded();
+
+        try {
+            final String embeddedJson = objectMapper.writeValueAsString(embedded);
+            return objectMapper.readValue(embeddedJson, EmbeddedAuthResponseDataV1.class);
+        } catch (IOException e) {
+            throw ApiException.newBuilder()
+                    .withApiErrors(DefaultApiError.SERVICE_UNAVAILABLE)
+                    .withExceptionCause(e)
+                    .withExceptionMessage("Error parsing the response body from OneLogin.")
+                    .build();
+        }
+    }
+
+    /**
+     * Print a user-friendly name for a MFA device
+     * @param factor - Okta MFA factor
+     * @return Device name
+     */
+    protected String getDeviceName(final Factor factor) {
+
+        Preconditions.checkArgument(factor != null, "factor cannot be null.");
+
+        final String factorProvider = factor.getProvider();
+        if (StringUtils.equalsIgnoreCase(factorProvider, "Google")) {
+            return "Google Authenticator";
+        }
+
+        return StringUtils.capitalize(factorProvider.toLowerCase());
```

Comment:
Why toLowerCase() and then capitalize?  Seems odd.  Mixed case seems nicer anyway, unless there is a reason to cap?